src/validations/access-checker/wearables.ts

```diff
@@ -275,6 +322,111 @@ export const wearables: Validation = {
+      // The keys used to create the hash MUST be present if they're required.
+      if (!MERKLE_PROOF_REQUIRED_KEYS.every((key) => merkleProof.hashingKeys.includes(key))) {
+        logs?.debug(`Merkle proof hashing keys don't satisfy the required keys`, {
+          requiredKeys: JSON.stringify(MERKLE_PROOF_REQUIRED_KEYS),
+          hashingKeys: JSON.stringify(merkleProof.hashingKeys),
+        })
+        return false
+      }
+      const generatedCrcHash = keccak256Hash(metadata, metadata.merkleProof.hashingKeys)
+      // The hash provided in the merkleProof for the entity MUST match the hash generated by the validator.
+      if (metadata.merkleProof.entityHash !== generatedCrcHash) {
+        logs?.debug(`The hash provided in the merkleProof doesn't match with the generated by the validator`, {
+          generatedCrcHash,
+          entityHash: metadata.merkleProof.entityHash,
+        })
+        return false
+      }
+
+      // Verify if the entity belongs to the Merkle Tree.
+      const bufferedProofs = merkleProof.proof.map((value) => toHexBuffer(value))
+      const bufferedMerkleRoot = toHexBuffer(merkleRoot)
+      return verifyProof(merkleProof.index, merkleProof.entityHash, bufferedProofs, bufferedMerkleRoot)
+    }
+
+    /**
+     * Verifies Third Party Wearable access by doing a Merkle Tree based verification, where the merkle root is stored in the blockchain
+     * and the proofs come in the wearable metadata.
+     * It also does a CRC using the metadata and some hashing keys:
+     *  - Hashing keys must contain all required keys defined in MERKLE_PROOF_REQUIRED_KEYS
+     *
+     * This checks are verified using blocks in a 5 minutes window from the deployment timestamp
+     */
+    const verifyMerkleProofedEntity = async (urn: BlockchainCollectionThirdParty): Promise<boolean> => {
+      // do merkle proofed entity validation, required keys are defined by the Catalyst and must be a subset of the hashingKeys from the MerkleProof to succeed
+      const metadata = deployment.entity.metadata as ThirdPartyWearable
+      const thirdPartyId = getThirdPartyId(urn)
+      const { blockNumberAtDeployment, blockNumberFiveMinBeforeDeployment } = await findBlocksForTimestamp(
+        externalCalls.subgraphs.L2.blocks,
+        deployment.entity.timestamp
+      )
+
+      let merkleRoots: string[] = []
+      const hasPermissionOnBlock = async (blockNumber: number | undefined) => {
+        if (!blockNumber) return false
+        const merkleRoot = await getMerkleRoot({
+          block: blockNumber,
+          thirdPartyId,
+        })
+        if (!merkleRoot) {
+          logs?.debug(`Merkle proof not found for given block and third party ID`, { blockNumber, thirdPartyId })
+          return false
+        }
+        merkleRoots.push(merkleRoot)
+        return await verifyHash(metadata, merkleRoot)
+      }
+
+      const validMerkleProofedEntity =
+        (await hasPermissionOnBlock(blockNumberAtDeployment)) ||
```

Comment:
can it happen that there are blocks in the middle missing?

Reply:
they will be missing, but I think it doesn't matter because it should not change at that frequency

what do you say @nachomazzara ?

Reply:
Blocks in the middle can't be missed